### PR TITLE
feat(Core/Misc): moved Position to its own file

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
##### CHANGES PROPOSED:
- Moved `struct Position` to its own file
- Part for dynamic linking

PR based TC commit https://github.com/TrinityCore/TrinityCore/commit/8a013ba0fbec8dd8dba8909f9fd149c2ec79c6a0

##### TESTS PERFORMED:
- Build and run in Windows at VS 16.3.8
- Test in game

##### HOW TO TEST THE CHANGES:
- Enter in game
- Try changing your position

##### KNOWN ISSUES AND TODO LIST:
- [x] Build in Travis
- [x] Test in game

##### Target branch(es):
- [x] Master

## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/game/Entities/Object/Position.h [hunk 1]
```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (C) 2016+     AzerothCore <www.azerothcore.org>, released under GNU GPL v2 license: https://github.com/azerothcore/azerothcore-wotlk/blob/master/LICENSE-GPL2
+ * Copyright (C) 2008-2016 TrinityCore <http://www.trinitycore.org/>
+ */
+
+#ifndef _POSITION_H_
+#define _POSITION_H_
+
+#include "Define.h"
+#include <string>
```
Comment (Viste) [inline src/server/game/Entities/Object/Position.h]: MapManager.h

File: src/server/game/Entities/Object/Position.cpp [hunk 2]
```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (C) 2016+     AzerothCore <www.azerothcore.org>
+ * Copyright (C) 2008-2015 TrinityCore <http://www.trinitycore.org/>
+ */
+
+#include "Position.h"
```
Comment (Viste) [inline src/server/game/Entities/Object/Position.cpp]: or #include "GridDefines.h"

**Comment by FrancescoBorzi** [comment]
build fails

File: src/server/game/Entities/Object/Position.cpp [hunk 3]
```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (C) 2016+     AzerothCore <www.azerothcore.org>
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Object/Position.cpp]: ``` * Copyright (C) 2016+     AzerothCore <www.azerothcore.org>, released under GNU GPL v2 license: https://github.com/azerothcore/azerothcore-wotlk/blob/master/LICENSE-GPL2```

Comment (Winfidonarleyan) [reply src/server/game/Entities/Object/Position.cpp]: This new file

Comment (FrancescoBorzi) [reply src/server/game/Entities/Object/Position.cpp]: but if it's copied from TC?

Comment (Winfidonarleyan) [reply src/server/game/Entities/Object/Position.cpp]: Not entirely, I made it myself, I used exactly the same file name, What do we do in that case?

Comment (FrancescoBorzi) [reply src/server/game/Entities/Object/Position.cpp]: ok, then  we can make it AGPL :+1:

**Pull request closed without merging**